wallet/src/de/schildbach/wallet/ui/more/SecurityFragment.kt

```diff
@@ -180,23 +183,44 @@ class SecurityFragment : Fragment(R.layout.fragment_security) {
             )
         } else {
             val resetWalletDialog = AdaptiveDialog.create(
-                null,
+                R.drawable.ic_warning_yellow_circle,
                 getString(R.string.reset_wallet_title),
                 getString(R.string.reset_wallet_message),
                 getString(R.string.button_cancel),
-                getString(R.string.positive_reset_text)
-            )
+                getString(R.string.reset_wallet_button)
+            ).apply {
+                requireArguments().putInt(AdaptiveDialog.POS_BUTTON_COLOR_ARG, R.style.PrimaryButtonTheme_Large_Red)
+            }
             resetWalletDialog.show(requireActivity()) {
                 if (it == true) {
-                    doReset()
+                    checkUsernameThenReset()
+                }
+            }
+        }
+    }
+
+    private fun checkUsernameThenReset() {
+        lifecycleScope.launch {
+            if (viewModel.hasIdentity && viewModel.hasPendingTxMetadataToSave()) {
+                SaveMetadataAndResetDialogFragment().show(requireActivity()) { result ->
+                    when (result) {
+                        null -> { }
+                        true, false -> {
+                            // TODO: Disable Reset Wallet for now
+                            // doReset()
+                        }
+                    }
                 }
+            } else {
+                // TODO: Disable Reset Wallet for now
+                // doReset()
```

Comment:
Should this be restored?

Reply:
Reset was disabled on purpose for this UI only PR.